app/controllers/api/v1/replies_controller.rb

```diff
@@ -23,4 +23,30 @@ def index
       .ordered
     paginate json: replies, per_page: per_page
   end
+
+  api :GET, '/replies/:id/bookmark', "Load a user's bookmark attached to a reply if it exists and is visible"
+  param :id, :number, required: true, desc: "Reply ID"
+  param :user_id, :number, required: true, desc: "User ID"
+  error 403, "Reply's post or user's bookmarks are not visible"
+  error 404, "Reply or user not found"
+  error 422, "Invalid parameters provided"
+  def bookmark
+    return unless (reply = find_object(Reply))
+    return unless (user = find_object(User, param: :user_id))
+    access_denied and return unless reply.post.visible_to?(current_user)
+
+    if !user.public_bookmarks && user.id != current_user.try(:id)
+      error = { message: "This user's bookmarks are private." }
+      render json: { errors: [error] }, status: :forbidden
+      return
+    end
+
+    bookmark = reply.bookmarks.find_by(user_id: user.id, type: "reply_bookmark")
+
+    if bookmark.present?
+      render json: bookmark.as_json
+    else
+      render json: {}
```

Comment:
I wonder if this would be better reported as an error or with a different status code. 404? If an end client needs to distinguish between the reply/user not existing and the bookmark not existing, they can parse the errors string.

Alternatively, if a reply can have _multiple_ bookmarks for a specific user (e.g. ones of different types), we could just return an empty array here and convert this into a pluralized endpoint?

Reply:
We can't have multiple bookmarks _yet_ and I think when we do I'll want to add a parameter to specify the type rather than have a list maybe? idk... that's fair though...